### DISCUSSION TITLE
permission

### DISCUSSION THREAD

**Discussion by nettle790** [codepharm/motiv8m3#76] [General]
hi sid

I have sorted out issue number #73,but when I pushed the changes the error reply was "permission not granted"to my repo.kindly check.

**Comment by IsthisSid** [comment]
Hi @nettle790 , sorry I'm just coming back from vacation. Thanks for working on this.
You'd have to create a pull request for code review before we can merge your branch into main. You can't directly push your changes.

After you've forked, cloned, and worked on the issue locally, create a new branch and name it something like 'add/73-link-lifestyle' and push your changes to your own remote repo. You should be then able to contribute to our repo and create a pull request. From there we will do a code review.

**Comment by nettle790** [comment]
Nice.Hope you had an enjoyable time on your vacation. I will follow the instructions accordingly.thanks for replying.

**Comment by IsthisSid** [comment]
Follow this guide [here](https://docs.github.com/en/get-started/quickstart/contributing-to-projects)
You're on this [step](https://docs.github.com/en/get-started/quickstart/contributing-to-projects#making-and-pushing-changes)

You'd be pushing your changes to your remote repo (playgroundpharma) which you forked from codepharm/playground.
Once that's done, you should see an option to contribute and open a pull request.

**Comment by nettle790** [comment]
There are lint errors in the file but I am not touching those , fixing only what you have mentioned in the issue.is <http://issue.is/> that fine?

**Reply by IsthisSid** [reply]
Just whatever is listed on the ticket is fine. Playground is just for all of us to practice so I'm sure we will run into issues that we'll fix together :)

**Comment by nettle790** [comment]
Ok.that is a relief.

**Comment by nettle790** [comment]
Attempted push again but error message saying your repo has denied permission to my repo.

**Reply by IsthisSid** [reply]
![image](https://user-images.githubusercontent.com/72266833/188933286-bb074757-995b-494a-950f-f647b4068478.png)
I don't see a new branch or that you've pushed your changes to your repo. You should be able to push changes to your remote repo that you forked. Once that's done, the banner will update and say you're however many commits ahead and be able to open a PR.

**Comment by nettle790** [comment]
My GUI is giving the error message.i have done all the steps you have mentioned before. Could it be the GUI issue???

**Reply by IsthisSid** [reply]
I'm unsure. I see that you've successfully done it before in one of your past repo's: 
![image](https://user-images.githubusercontent.com/72266833/188935746-9629f76d-6824-4e6b-8aad-a04ef70bf1e0.png)

Here you created a feature1 branch and have 1 commit ahead and able to open a PR. I'll test it out on my end with another collaborator and see if I run into any issues as well.

**Comment by nettle790** [comment]
I checked , GUI is fine.For some reason your repo is not granting permission
I solved your issue  a few days ago but when I attempted to push it ,it failed citing the same error message.

**Reply by codeTrackLift** [reply]
Hello @nettle790, sorry about the difficulties.  Let's try this in our Git Practice repo....  

@nettle790 Added you as a contributor with write privileges to https://github.com/codepharm/git

**Comment by nettle790** [comment]
Ok

**Comment by nettle790** [comment]
Sid
 After I accepted the invite from Pete it showed push access enabled to codepharm/git.
You haven’t granted me access!!!!!!

**Reply by codeTrackLift** [reply]
@nettle790 we are just starting to allow outside contributors, please be patient and respectful.

**Comment by nettle790** [comment]
Thanks for the invite.my apologies if any part of my replies have offended.

**Reply by IsthisSid** [reply]
No offense taken. You were trying to push to our main and the only thing Pete did was allow you access to write changes in our git practice repo. 

However, as an outside collaborator to the Playground repo, you have to fork and submit a PR. 
I just had someone test it out as an outside collaborator and it worked. There is now a PR in the playground repo for me to code review if you wanted to check out the steps he took to get there.

**Reply by mdwstrat** [reply]
<img width="912" alt="Screen Shot 2022-09-07 at 2 49 36 PM" src="https://user-images.githubusercontent.com/28866111/188955269-096a0a9f-db05-4a7b-b06c-38a8145c8875.png">

**Comment by nettle790** [comment]
Sid please check the changes. I have pushed them.
